middleware.js

```diff
@@ -13,6 +13,47 @@ export function middleware (request) {
   return resp
 }
 
-export const config = {
-  matcher: ['/(.*/|)r/([\\w_]+)([?#]?.*)']
+const multiAuthMiddleware = (request) => {
+  // switch next-auth session cookie with multi_auth cookie if cookie pointer present
+
+  // is there a cookie pointer?
+  const cookiePointerName = 'multi_auth.user-id'
+  const hasCookiePointer = request.cookies?.has(cookiePointerName)
+  // is there a session?
+  const sessionCookieName = '__Secure-next-auth.session-token'
+  const hasSession = request.cookies?.has(sessionCookieName)
+
+  if (!hasCookiePointer || !hasSession) {
+    // no session or no cookie pointer. do nothing.
+    return NextResponse.next({ request })
+  }
+
+  const userId = request.cookies?.get(cookiePointerName)?.value
+  if (userId === 'anonymous') {
+    // user switched to anon. only delete session cookie.
+    request.cookies.delete(sessionCookieName)
+    return NextResponse.next({ request })
+  }
+
+  const userJWT = request.cookies.get(`multi_auth.${userId}`)?.value
+  if (!userJWT) {
+    // no multi auth JWT found
+    return NextResponse.next({ request })
+  }
+
+  if (userJWT) {
+    // multi auth JWT found in cookie that pointed to by cookie pointer that is different to current session cookie.
+    request.cookies.set(sessionCookieName, userJWT)
+    return NextResponse.next({ request })
+  }
+
+  return NextResponse.next({ request })
+}
```

Comment:
You say in the PR comment

> On click, another cookie multi_auth.user-id is created via JS. This cookie is read inside a middleware in the backend to replace the session cookie that will be used to determine the user in the backend.

Are we not able to make this switch, replacing the session cookie, on the client? Just want to make sure we *need* this middleware.

Reply:
> Are we not able to make this switch, replacing the session cookie, on the client? Just want to make sure we need this middleware.

No since it's not accessible from JS